rasa/importers/autoconfig.py

```diff
@@ -0,0 +1,197 @@
+import copy
+import logging
+import os
+import re
+import shutil
+from pathlib import Path
+import tempfile
+
+from typing import Text, Dict, Any, List, Set
+
+from rasa.cli import utils as cli_utils
+from rasa.constants import (
+    CONFIG_AUTOCONFIGURABLE_KEYS,
+    DOCS_URL_PIPELINE,
+    DOCS_URL_POLICIES,
+    CONFIG_KEYS,
+)
+from rasa.utils import io as io_utils
+
+logger = logging.getLogger(__name__)
+
+
+# In future iterations, this function and the function `create_config_for_keys` below
+# will get additional input parameters to allow for smarter configuration. E.g.
+# training_data: TrainingData
+# domain: Domain
+# stories: StoryGraph
+def get_auto_configuration(config_file) -> Dict[Text, Any]:
+    """Determine configuration from a configuration file.
+
+    Keys that are provided in the file are kept. Keys that are not provided are
+    configured automatically.
+    """
+    if config_file and os.path.exists(config_file):
+        config = io_utils.read_config_file(config_file)
+
+        missing_keys = [k for k in CONFIG_AUTOCONFIGURABLE_KEYS if not config.get(k)]
+        config = _create_config_for_keys(config, missing_keys)
+
+        dump_config(config, config_file)
+    else:
+        config = {}
+
+    return config
+
+
+def _create_config_for_keys(
```

Comment:
Good point. I will also rename `get_auto_configuration` to just `get_configuration` then, because automatic configuration is not always involved, only when there are keys missing.